### PR TITLE
AC-5181: Improve deploy documentation for impact-api

### PR DESCRIPTION
**Pull request by mc-nathan**
To test:
- Follow the instructions under "Deploying to an Environment"

### CODE REVIEW

**Review by yotammanor — CHANGES_REQUESTED** [review]
Getting a timeout every time I try to deploy (to master, environment staging, registry is correct).

File: README.md [hunk 1]
```diff
@@ -262,15 +271,27 @@ date/release tag for the deploy and then committed to the branch in
 question.  This mean commiting directly to the target branch such as
 'master' or 'AC-9999'.
 
-Manage your secret keys and other sensitive information through environment variables. The MassChallenge team manages our environment variables in our Continuous Integration service, Travis. :More details can be found [here](https://docs.travis-ci.com/user/environment-variables/).
+Manage your secret keys and other sensitive information through
```
[the remark this answers is not in the document]
Comment (mc-nathan) [reply README.md]: You are correct.  Updated.

File: README.md [hunk 2]
```diff
@@ -291,14 +307,18 @@ Deploying task definition.......................................................
 Deployment successful
 ```
 
-Note - if the IMAGE_TAG argument is not provided to the deploy command, the name of the current branch is used as the IMAGE_TAG
+Note - if the IMAGE_TAG argument is not provided to the deploy
+command, the name of the current branch is used as the IMAGE_TAG
 
-You can also check that your environment is functioning after a successful deployment 
-by running the following commands without problems on the master branch:
+You can also check that your environment is functioning after a
```
Comment (pbx) [inline README.md]: Confusing. Is the implication that my local environment might be broken by the deploy somehow? If not, why is this with the deploy instructions?

**Review by pbx — CHANGES_REQUESTED** [review]
Good improvements. Flagged a few things i  I think need attention. 

As for doing a successful deploy, I suffered the "Deployment failed due to timeout" problem like Yotam.  I added a `--timeout` value to the `ecs deploy` command in the Makefile's `deploy` target, but **still got a failure because of timeout even when it ran for an hour (`--timeout=3600`).**

File: README.md (hunk 2 above)
Comment (mc-nathan) [reply README.md]: Deleted.

File: README.md [hunk 3]
```diff
@@ -211,66 +212,81 @@ Example:
...
+
+An example of this to deploy current master to staging looks like this:
+```
+DOCKER_REGISTRY=`aws ecr describe-repositories --repository-name impact-api | cut -d"\"" -f4 | cut -d"/" -f1 | grep -i "amazonaws.com"`
+make deploy IMAGE_TAG=master ENVIRONMENT=staging DOCKER_REGISTRY=$DOCKER_REGISTRY
+```
 
-(NOTE - The value for the DOCKER_REGISTRY can be grabbed from the ecr page (https://console.aws.amazon.com/ecs/home?region=us-east-1#/repositories/impact-api#images;tagStatus=ALL) listed as the "Repository URI" for the impact-api image or from running the following bash command ```aws ecr  describe-repositories --repository-name impact-api  | cut -d"\"" -f4 | cut -d"/" -f1  | grep -i "amazonaws.com"```)
+The value for DOCKER_REGISTRY is expected to be a host name typically
+ending with `.amazonaws.com`.  The value of the host should be given
```
[the remark this answers is not in the document]
Comment (mc-nathan) [reply README.md]: Fixed (but edited slightly).

File: README.md [hunk 4]
```diff
@@ -211,44 +212,52 @@ Example:
...
+The `make grant-permissions` command can be used to assign django
+permissions for a given user like so:
 
 `make grant-permissions PERMISSION_CLASSES=view_startup,change_startup PERMISSION_USER=test@example.org`
 
-To give a user access to the v0 api, issue the command with 'v0_clients' specified in PERMISSION_CLASSES:
+To give a user access to the v0 api, issue the command with
+'v0_clients' specified in PERMISSION_CLASSES:
 
 `make grant-permissions PERMISSION_USER=test@example.org PERMISSION_CLASSES=v0_api`
```
[the remark this answers is not in the document]
Comment (mc-nathan) [reply README.md]: Fixed.

**Review by pbx — APPROVED** [review]

**Comment by pbx** [comment]
Deploying works now.

**Comment by yotammanor** [comment]
Works for me to. I'd add a reference to the versions of aws-cli and ecs-deploy that are not failing (awscli==1.11.75 and ecs-deploy==1.4.3 for me), and create a follow on ticket that checks why the problem we had occurred.

**Comment by mc-nathan** [comment]
Added current versions numbers based on what I saw and what Yotam reported.  Added ticket AC-5199.  Can we get this bug merged and done????

**Review by yotammanor — APPROVED** [review]

**Pull request merged**